src/master/filesystem_operations.cc

```diff
@@ -2158,6 +2158,53 @@ uint8_t fs_apply_incversion(uint64_t chunkid) {
 	return chunk_increase_version(chunkid);
 }
 
+#ifndef METARESTORE
+uint8_t fs_remove_chunk_from_file(const FsContext &context, inode_t inode, uint64_t chunkId) {
+	uint32_t ts = eventloop_time();
+	ChecksumUpdater cu(ts);
+	StatsRecord psr, nsr;
+	FSNode *p;
+
+	if (chunkId == 0) { return SAUNAFS_ERROR_NOCHUNK; }
+
+	uint8_t status = verify_session(context, OperationMode::kReadWrite, SessionType::kAny);
+	if (status != SAUNAFS_STATUS_OK) { return status; }
+
+	status =
+	    fsnodes_get_node_for_operation(context, ExpectedNodeType::kFile, MODE_MASK_W, inode, &p);
+	if (status != SAUNAFS_STATUS_OK) { return status; }
+
+	auto *node_file = dynamic_cast<FSNodeFile *>(p);
+	fsnodes_get_stats(p, &psr);
+	uint32_t indx = 0;
+	for (; indx < node_file->chunks.size(); indx++) {
+		if (node_file->chunks[indx] == chunkId) { break; }
+	}
+
+	// not found
+	if (indx == node_file->chunks.size()) { return SAUNAFS_ERROR_NOCHUNK; }
+
+	status = chunk_delete_file(chunkId, p->goal);
+	node_file->chunks[indx] = 0;
+	p->mtime = ts;
+	fsnodes_update_ctime(p, ts);
+	if (status != SAUNAFS_STATUS_OK) { return status; }
```

Comment:
@lgsilva3087 you mean to MOVE check above immediately after we got status?
 ```suggestion
	status = chunk_delete_file(chunkId, p->goal);
	if (status != SAUNAFS_STATUS_OK) { return status; }
	node_file->chunks[indx] = 0;
	p->mtime = ts;
	fsnodes_update_ctime(p, ts);
```

Reply:
Yes, that's what he meant.